SourceX/DiabloUI/diabloui.cpp

```diff
@@ -324,7 +336,7 @@ void UiFocusNavigation(SDL_Event *event)
 #ifndef USE_SDL1
 		case SDL_TEXTINPUT:
 			if (textInputActive) {
-				selhero_CatToName(event->text.text, UiTextInput, UiTextInputLen);
+				selhero_SetName(event->text.text, UiTextInput, UiTextInputLen);

```

Comment:
This won't do the right thing if you're typing on a physical keyboard, which sends an event for each "letter"

Reply:
Ugh. So i guess this should be done for vita only. What about switch?

Reply:
Perhaps we could do something like:

```c++
if (SDL_HasScreenKeyboardSupport())
  selhero_SetName(event->text.text, UiTextInput, UiTextInputLen);
else
  selhero_CatToName(event->text.text, UiTextInput, UiTextInputLen);
```

Reply:
That'll probably break on ios/android/winphone, as those do have screen keyboard support, but also produce events for each key press.

Reply:
In that case we have to ifdef here it seems

Reply:
> What about switch?

I don't know but however it was set up before worked (at least there are no bug reports about this)

Reply:
It looks like this abomination https://github.com/diasurgical/devilutionX/blob/master/SourceX/platform/switch/keyboard.cpp#L55-L62
deals with prefiled text on switch.
So i've added ifdefs for both platforms and code above shouldn't be required anymore.

Reply:
The Switch still currently sends individual events for each code point:

https://github.com/diasurgical/devilutionX/blob/eeccf6b5d6c8f11e9e45b93845d3a5e2aadefe5b/SourceX/platform/switch/keyboard.cpp#L68-L71

So we'll receive individual code points and they will each overwrite the text.

There is also the 3ds

Reply:
Oh, god dammit. Ok, i've removed switch ifdef, but that should really be fixed in switch implementation

Reply:
And 3ds, looks like, does OSK stuff even before name dialog. And, well, it's SDL1.
So, well, after fixing that mess in switch code, that ifdef for switch can be returned.